src/Microsoft.Azure.WebJobs/ParameterBindingData.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// A reference type for supporting SDK-type bindings in out-of-process Function workers
+    /// </summary>
+    public class ParameterBindingData
+    {
+        /// <summary> Initializes a new instance of the <see cref="ParameterBindingData"/> class with the Content-Type set to application/json</summary>
+        /// <param name="source"> Identifies the extension this event is coming from </param>
+        /// <param name="jsonSerializableData"> An object containing any required information to hydrate an SDK-type object in the out-of-process worker </param>
+        /// <param name="dataSerializationType"> The type to use when serializing the data.
+        /// If not specified, <see cref="object.GetType()"/> will be used on <paramref name="jsonSerializableData"/>.</param>
+        /// <exception cref="ArgumentNullException">Throws when source is null.</exception>
+        /// <exception cref="ArgumentException">Throws when jsonSerializableData is of type BinaryData.</exception>
+        public ParameterBindingData(string source, object jsonSerializableData, Type dataSerializationType = default)
+        {
+            if (jsonSerializableData is BinaryData)
+            {
+                throw new ArgumentException("This constructor does not support BinaryData. Use the constructor that takes a BinaryData instance.");
+            }
+
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            Version = "1.0";
+            Source = source;
+            ContentType = "application/json";
+            Content = new BinaryData(jsonSerializableData, type: dataSerializationType ?? jsonSerializableData?.GetType());
+        }
+
+        /// <summary> Initializes a new instance of the <see cref="ParameterBindingData"/> class using binary event data.</summary>
```

Comment:
nit: it's not really "event data". Perhaps just simplify this to "Initializes a new instance of the ParameterBindingData class using the specified BindingData and content type."